### ISSUE TITLE
REBALANCE hint produces unexpected spacing error in spark SQL

### ISSUE THREAD

**Issue by barunpuri** [sqlfluff/sqlfluff#5869]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

REBALANCE hint produces unexpected spacing error in spark SQL
other hints (with brackets) pass as expected

### Expected Behaviour

Pass without error

### Observed Behaviour

linting failed with error 
```
== [stdin] FAIL
L:   2 | P:  21 | LT01 | Unexpected whitespace before end hint '*/'.
                       | [layout.spacing]
All Finished 📜 🎉!
```

### How to reproduce

`echo 'INSERT OVERWRITE TABLE aaa.bbb PARTITION (a)\nSELECT /*+ REBALANCE */ 1 AS a FROM xxx.yyy' | sqlfluff lint --dialect sparksql -`

### Dialect

sparksql

### Version

3.0.6

### Configuration

default settings

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
